src/openforms/submissions/form_logic.py

```diff
@@ -107,7 +113,11 @@ def evaluate_form_logic(
         data_diff = {}
         for key, new_value in step.data.items():
             original_value = data.get(key)
-            if new_value == original_value:
+            # Reset the value of any field that may have become hidden again after evaluating the logic
+            if original_value and get_component(configuration, key).get("hidden"):
+                data_diff[key] = defaults.get(key, "")
+                continue
```

Comment:
shouldn't this only apply if `clearOnHide` is set?

Reply:
Good point